payments-core/src/main/java/com/stripe/android/IntentConfirmationInterceptor.kt

```diff
@@ -0,0 +1,226 @@
+                        error = IllegalStateException(
+                            "ConfirmCallback must be implemented when using " +
+                                "IntentConfiguration to configure PaymentSheet",
+                        ),
+                    )
+                }
+            }
+        }
+    }
+
+    private suspend fun createPaymentMethod(params: PaymentMethodCreateParams): PaymentMethod? {
+        return runCatching {
+            stripeRepository.createPaymentMethod(
+                paymentMethodCreateParams = params,
+                options = requestOptions,
+            )
+        }.getOrNull()
+    }
+
+    private suspend fun handleClientSideConfirmation(
+        createIntentCallback: CreateIntentCallback,
+        paymentMethod: PaymentMethod,
+        shippingValues: ConfirmPaymentIntentParams.Shipping?
+    ): IntentConfirmationInterceptor.NextStep {
+        val result = createIntentCallback.onCreateIntent(paymentMethodId = paymentMethod.id!!)
+
+        return when (result) {
+            is CreateIntentCallback.Result.Success -> {
+                createConfirmStep(result.clientSecret, shippingValues, paymentMethod)
+            }
+            is CreateIntentCallback.Result.Failure -> {
+                IntentConfirmationInterceptor.NextStep.Fail(result.error)
+            }
+        }
+    }
+
+    private suspend fun handleServerSideConfirmation(
+        createIntentCallback: CreateIntentCallbackForServerSideConfirmation,
+        paymentMethod: PaymentMethod,
+        shouldSavePaymentMethod: Boolean,
+        shippingValues: ConfirmPaymentIntentParams.Shipping?,
+    ): IntentConfirmationInterceptor.NextStep {
+        val result = createIntentCallback.onCreateIntent(
+            paymentMethodId = paymentMethod.id!!,
+            shouldSavePaymentMethod = shouldSavePaymentMethod
+        )
+
+        return when (result) {
+            is CreateIntentCallback.Result.Success -> {
+                val intent = retrieveStripeIntent(result.clientSecret)
+
+                if (intent?.isConfirmed == true) {
+                    IntentConfirmationInterceptor.NextStep.Complete
+                } else if (intent?.status == StripeIntent.Status.RequiresAction) {
+                    IntentConfirmationInterceptor.NextStep.HandleNextAction(result.clientSecret)
+                } else {
+                    createConfirmStep(result.clientSecret, shippingValues, paymentMethod)
+                }
+            }
+            is CreateIntentCallback.Result.Failure -> {
+                IntentConfirmationInterceptor.NextStep.Fail(result.error)
+            }
+        }
+    }
+
+    private suspend fun retrieveStripeIntent(clientSecret: String): StripeIntent? {
+        return runCatching {
+            stripeRepository.retrieveStripeIntent(
+                clientSecret = clientSecret,
+                options = requestOptions,
+            )
+        }.getOrNull()
```

Comment:
Are we essentially swallowing this error?

Reply:
Same as above: We do right now, but we can change that.